src/main.c

```diff
@@ -385,9 +392,39 @@ static void library_main_helper(struct libargs_s *args) {
                 handle_check_address(args->check_address, args->coin_config);
             break;
         case SIGN_TRANSACTION:
+            initialize_app_globals();
             if (copy_transaction_parameters(args->create_transaction)) {
                 // never returns
-                handle_swap_sign_transaction(args->coin_config);
+
+                G_coin_config = args->coin_config;
+#ifndef DISABLE_LEGACY_SUPPORT
+                // We make sure to initialize the app in "legacy" mode, otherwise the state
+                // would be wiped in app_main
+                memset(&btchip_context_D, 0, sizeof(btchip_context_D));
+                btchip_context_init();
+                G_app_mode = APP_MODE_LEGACY;
+#else
+                G_app_mode = APP_MODE_UNINITIALIZED;
+#endif
+                G_swap_state.called_from_swap = 1;
+
+                io_seproxyhal_init();
+                UX_INIT();
+                ux_stack_push();
+
+                USB_power(0);
+                USB_power(1);
+                // ui_idle();
+                PRINTF("USB power ON/OFF\n");
+#ifdef TARGET_NANOX
```

Comment:
could be HAVE_BLE

Reply:
Leaving as is for now, as it's how the app-boilerplate does